core/test/element_arithmetic_test.cpp

```diff
@@ -76,6 +76,31 @@ TEST(ElementArithmeticIntegerDivisionTest, inplace_truediv_not_supported) {
   EXPECT_TRUE(no_int_as_first_arg(decltype(divide_equals)::types{}));
 }
 
+TEST(ElementArithmeticIntegerDivisionTest, mod) {
+  EXPECT_EQ(mod(-3, 3), 0);
+  EXPECT_EQ(mod(-2, 3), 1);
+  EXPECT_EQ(mod(-1, 3), 2);
+  EXPECT_EQ(mod(0, 3), 0);
+  EXPECT_EQ(mod(1, 3), 1);
+  EXPECT_EQ(mod(2, 3), 2);
+  EXPECT_EQ(mod(3, 3), 0);
+  EXPECT_EQ(mod(4, 3), 1);
+}
+
+TEST(ElementArithmeticIntegerDivisionTest, mod_equals) {
+  constexpr auto check_mod = [](auto a, auto b, auto expected) {
+    mod_equals(a, b);
+    EXPECT_EQ(a, expected);
+  };
+  check_mod(-2, 3, 1);
+  check_mod(-1, 3, 2);
+  check_mod(0, 3, 0);
+  check_mod(1, 3, 1);
+  check_mod(2, 3, 2);
+  check_mod(3, 3, 0);
+  check_mod(4, 3, 1);
+}
+
```

Comment:
Missing devisor < 0 with dividend >0, =0, <0. Also cases of `|dividend| > |divisor|` with either / both operands negative.

Reply:
@jl-wynen Done. Was not sure how to handle mod 0. Without anything it just crashes, so I return 0 instead (which is what numpy seems to do). Numpy also prints a warning, but right now we don't have a mechanism for that in `transform`. One option could be to accumulate with an extra flag argument, but this would make operation slower (since not multi-threaded), so in any case I would leave that for another issue/PR, in particular since it also affects division.

Reply:
Python raises `ZeroDivisionError` and not just a warning. Not sure if doing this can have any impact on performance.

Tests still missing 
```.cpp
EXPECT_EQ(mod(-4, 3), 2);
```
and
```.cpp
check_mod(-4, 3, 2);
check_mod(-3, 3, 0);
```
I know, I'm very fussy about it :-)

Reply:
> Python raises `ZeroDivisionError` and not just a warning. Not sure if doing this can have any impact on performance.

I suppose numpy does not do this since it would mean that a single 0 in an array would prevent the operation for the entire array?

Reply:
But how do you use the result when you don't know if something went wrong? A warning is only useful in an interactive context. Or maybe returning 0 makes the using code well behaved in almost all contexts?

Reply:
I don't know, would be interesting how numpy decided to do what they do?